### ISSUE TITLE
Files and images as block attachments

### ISSUE THREAD

**Issue by tiberiuichim** [plone/volto#1561]
It's looking more obvious that blocks can be their own mini-applications: they're a contact form, an image gallery, a listing, a map, etc. They stand more and more on their own, just like a separate content type. 

So just like a content type has individual file fields, I argue that the blocks should be able to "embed" their own files. I've been exploring this concept and have a WIP implementation where file and images are stored (as blobs, of course) in the annotation of the context and they can be referenced by the block.

**Comment by tisto** [comment]
@tiberiuichim I fully agree with your opinion here. Blocks are the core building blocks of Volto and we will have put some effort into getting this right. Somehow I have the feeling that we are re-inventing something new on top of the existing layers we have and we are solving the same problems over and over again. So I am +1 discussing this but it won't be an easy one.

**Comment by tiberiuichim** [comment]
@tisto I still see a lot of value in regular Plone content types, to stay as they are now. They have workflow + security, indexing, publishing (their own URLs), etc. So from my perspective the Plone content types become containers for the block mini-applications. To have the Volto blocks exposed as Plone content items wouldn't make sense (IMHO).

To a certain degree, how blocks are stored in the backend is not that important. But a page composition solution such as https://github.com/collective/collective.frontpage feels "ugly": a lot of special content is created, just to fit the limitations of the Plone platform.

**Comment by cekk** [comment]
@tiberiuichim this discussion is also about the issue that you showed us during the sprint?
Right now, with folderish contents when we are in creation form and add an image block with a new uploaded image, that image is created in its container (because you haven't created it yet).
But when you edit it and add a new image block, the new image is created inside it.
This can be confusing for editors because they have two images loaded in two different places.

With your proposal this could be fixed, but we should still allow users to reference an already created Image content in the portal

**Comment by tiberiuichim** [comment]
@cekk It would be another approach to solve that image creation issue. 

But it's also about harmonization of concepts inside Volto: with blocks we're unshackling the content from the bounds imposed by the content types. If I can create any block in a content and it wouldn't appear anywhere else in the portal, why can't I add a file attachment to that content?

**Comment by cekk** [comment]
probably we should give both options.
Right now in plone you can:
- upload an image directly in the content-type (with the leadimage behavior field for example)
- reference one Image content in TinyMCE field

The first case could be still managed adding image fields to the content types when needed.

Blocks are used more or less to replace the rich text field (with blocks we are building the body of the page), so for me it's not a common use-case the reuse of images for editors. But there should also be the use-case where for a page they need a one-shot image and they don't want to upload it in a common folder or inside that content, but it's ok to store it into the block.
